File: 001. Two Sum/solution.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> ans;
        unordered_map<int, int> mp;
        for (int i = 0; i < nums.size(); i++) {
            if (mp.find(target - nums[i]) != mp.end()) {
                ans = { i, mp[target - nums[i]] };
                return ans;
            }
            mp[nums[i]] = i;
        }
        return ans;
    }
};

void print(std::vector <int> const& a) {
    std::cout << "The answer is: ";

    for (int i = 0; i < a.size(); i++)
        std::cout << a.at(i) << ' ';
}

int main()
{
    vector<int> nums;
    int target = 18;

    nums.push_back(2);
    nums.push_back(7);
    nums.push_back(11);
    nums.push_back(15);


    Solution solution;

    print(solution.twoSum(nums, target));
  
    return 0;
}

File: 001. Two Sum/solution.java
import java.util.HashMap;
import java.util.Scanner;
import java.util.Map;
import java.util.Arrays;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[j] == target - nums[i]) {
                    return new int[] { i, j };
                }
            }
        }
        // In case there is no solution, we'll just return null
        return null;
    }
}


public class main {
	public static void main(String[] args) {
		Solution solution = new Solution();
		int[] nums;
		nums = new int[4];
		nums[0] = 2;
		nums[1] = 7;
		nums[2] = 11;
		nums[3] = 15;
		int target = 9;
	    System.out.println("The answer is:");
	    System.out.println(Arrays.toString(solution.twoSum(nums, target)));
	}
}
